### PR TITLE
Support GPT partition tables

### PR DESCRIPTION
**Pull request by zvin**
This changes the return value of getPartitions()
Instead of being a list of partitions, it will now be an object with:
 * type: 'mbr' or 'gpt'
 * partitions: the list of partitions as returned before

Connects-To: #25

Change-Type: major
---- Autogenerated Waffleboard Connection: Connects to #25

### CODE REVIEW

**Comment by jviotti** [comment]
@zvin Just as a side note: Can you add @abrodersen as a reviewer for partition related stuff, given the Reconfix work he's doing?

File: lib/partitioninfo.coffee [hunk 1]
```diff
@@ -1,42 +1,57 @@
...
+GPT = require('gpt')
 Promise = require('bluebird')
 
 ###*
 # @module partitioninfo
 ###
 
-BOOT_RECORD_SIZE = 512
+MBR_SIZE = 512
+GPT_SIZE = 512 * 33
```
Comment (jviotti) [inline lib/partitioninfo.coffee:12]: Is there a way we can get these constants from the `gpt` or `mbr` module? If they are not there, can we add them? @jhermsmeier ?

**Review by jviotti — COMMENTED** [review]
Not sure if its relevant, but it looks like we have very similar code in Etcher (that attempts to parse MBR or GPT). If so, can we re-use the code somehow? @jhermsmeier should we move more functionality into your partition table modules, or potentially create a master one that uses `gpt` and `mbr` under the hood?

**Comment by zvin** [comment]
@jviotti ok I will. This PR doesn't do anything fancy, it just uses node-gpt when the disk image appears to be using gpt.

File: lib/partitioninfo.coffee (hunk 1 above)
Comment (zvin) [reply lib/partitioninfo.coffee:12]: It looks like mbr and gpt do not export anything like this.

File: lib/partitioninfo.coffee [hunk 2]
```diff
@@ -54,22 +69,34 @@ getLogicalPartitions = (disk, index, offset, extendedPartitionOffset = offset, l
...
-			else
-				result.push(partitionDict(p, options.offset, index + 1))
-		if extended != null and options.getLogical
-			return getLogicalPartitions(disk, MBR_FIRST_LOGICAL_PARTITION, extended.byteOffset())
-			.then (logicalPartitions) ->
-				result.push(logicalPartitions...)
+	readFromDisk(disk, options.offset, MBR_SIZE)
+	.then (mbrBuf) ->
+		partitions = getPartitionsFromMBRBuf(mbrBuf)
+		if partitions.length == 1 and partitions[0].type == GPT_PROTECTIVE_MBR
```
Comment (CameronDiver) [inline lib/partitioninfo.coffee]: Super NIT: Can we have these as `is`? I find it a lot more readable, and we use it elsewhere.

File: lib/partitioninfo.coffee [hunk 3]
```diff
@@ -54,22 +69,34 @@ getLogicalPartitions = (disk, index, offset, extendedPartitionOffset = offset, l
...
+			result.partitions = []
+			result.type = 'mbr'
+			for p, index in partitions
+				if MBR.Partition.isExtended(p.type)
+					extended = p
+					if options.includeExtended
+						result.partitions.push(mbrPartitionDict(p, options.offset, index + 1))
+				else
+					result.partitions.push(mbrPartitionDict(p, options.offset, index + 1))
+			if extended != null and options.getLogical
```
Comment (CameronDiver) [inline lib/partitioninfo.coffee]:
```
if extended? and options.getLogical

File: lib/partitioninfo.coffee [hunk 4]
```diff
@@ -79,14 +106,19 @@ get = (disk, number) ->
 	if number < 1
 		throw new Error('The partition number must be at least 1.')
 	getPartitions(disk, includeExtended: true, offset: 0, getLogical: false)
-	.then (partitions) ->
+	.then (info) ->
+		if info.type == 'gpt'
```
Comment (CameronDiver) [inline lib/partitioninfo.coffee]: `info.type is 'gpt'`

**Review by CameronDiver — APPROVED** [review]

File: lib/partitioninfo.coffee (hunk 4 above)
Comment (zvin) [reply lib/partitioninfo.coffee]: done

File: lib/partitioninfo.coffee (hunk 3 above)
Comment (zvin) [reply lib/partitioninfo.coffee]: updated

File: lib/partitioninfo.coffee (hunk 2 above)
Comment (zvin) [reply lib/partitioninfo.coffee]: udpated

**Pull request merged**